### PR TITLE
treeview.py: remove legacy verify_grouping_mode() function

### PR DESCRIPTION
**Pull request by slook**
- Removed: Legacy function in favor of permanent config migration.

### CODE REVIEW

File: pynicotine/config.py [hunk 1]
```diff
@@ -699,6 +699,30 @@ def _set_config(self):
...
+
+                    elif mode == "2":
+                        mode = "user_grouping"
+
+                    # Verify mode validity
+                    elif mode not in {"ungrouped", "folder_grouping", "user_grouping"}:
+                        mode = "folder_grouping"
+
+                    self.sections[section][option] = mode
+                    continue
```
Comment (mathiascode) [inline pynicotine/config.py:724]: This code block isn't currently reached. It should be moved down after line 728 (change `if (section, option) in [...]` to `for section, option in (...)`).

File: pynicotine/config.py [hunk 2]
```diff
@@ -699,6 +699,30 @@ def _set_config(self):
...
+
+                    elif mode == "1":
+                        mode = "folder_grouping"
+
+                    elif mode == "2":
+                        mode = "user_grouping"
+
+                    # Verify mode validity
+                    elif mode not in {"ungrouped", "folder_grouping", "user_grouping"}:
+                        mode = "folder_grouping"
```
Comment (mathiascode) [inline pynicotine/config.py:721]: We can remove this code. We should only take care of migrating old config values.

**Review by mathiascode — CHANGES_REQUESTED** [review]

File: pynicotine/config.py (hunk 1 above)
Comment (slook) [reply pynicotine/config.py]: Oh dear, the method has become too complex. Perhaps we need a dedicated `_migrate_config()` function, or such?

Comment (mathiascode) [reply pynicotine/config.py]: Most likely, but don't include it in this PR.

File: pynicotine/config.py [hunk 3]
```diff
@@ -702,6 +702,26 @@ def _set_config(self):
                 # Set default value
                 self.sections[section][option] = value
 
+        # Map legacy folder/user grouping modes (3.1.0)
+        for (section, option) in [
+            ("searches", "group_searches"),
+            ("transfers", "groupdownloads"),
+            ("transfers", "groupuploads")
+        ]:
```
Comment (mathiascode) [inline pynicotine/config.py:710]:
```suggestion
        for section, option in (
            ("searches", "group_searches"),
            ("transfers", "groupdownloads"),
            ("transfers", "groupuploads")
        ):
```

Small nit. This is the code style we use elsewhere nowadays.

File: pynicotine/config.py (hunk 1 above)
Comment (mathiascode) [reply pynicotine/config.py]: Looks like the linter agrees too

**Review by mathiascode — APPROVED** [review]

File: pynicotine/config.py (hunk 1 above)
Comment (mathiascode) [reply pynicotine/config.py]: Done in 778e2748d8ace661c2549eaf973a985f1d65162b

**Pull request merged**